src/pyorderly/outpack/location.py

```diff
@@ -169,3 +178,35 @@ def _location_driver(location_name, root) -> LocationDriver:
 
     msg = "invalid location type"
     raise Exception(msg)
+
+
+def _find_all_dependencies(
+    packet_ids: list[str],
+    metadata: dict[str, MetadataCore],
+    *,
+    allow_missing_packets: bool = False,
+) -> list[str]:
+    result = []
+
+    # This is a standard breath first search through the packet graph.
```

Comment:
is this not depth first? youre appending and popping from the same side like a stack
```suggestion
    # This is a standard depth first search through the packet graph.
```

Reply:
Yes good catch. I've also extended the comment below about ordering of the result.